repairnator/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/process/inspectors/metrics4bears/TestMetrics4BearsJsonFile.java

```diff
@@ -0,0 +1,435 @@
+        config.setPush(true);
+        config.setPushRemoteRepo("");
+
+        String solverPath;
+        if (isMac()) {
+            solverPath = SOLVER_PATH_DIR+SOLVER_NAME_MAC;
+        } else {
+            solverPath = SOLVER_PATH_DIR+SOLVER_NAME_LINUX;
+        }
+        config.setZ3solverPath(solverPath);
+
+        jsonMapper = new ObjectMapper();
+        String workingDir = System.getProperty("user.dir");
+        workingDir = workingDir.substring(0, workingDir.lastIndexOf("repairnator/"));
+        String jsonSchemaFilePath = workingDir + "resources/bears-schema.json";
+        File jsonSchemaFile = new File(jsonSchemaFilePath);
+        JsonNode schemaObject = jsonMapper.readTree(jsonSchemaFile);
+
+        LoadingConfiguration loadingConfiguration = LoadingConfiguration.newBuilder().dereferencing(Dereferencing.INLINE).freeze();
+        JsonSchemaFactory factory = JsonSchemaFactory.newBuilder().setLoadingConfiguration(loadingConfiguration).freeze();
+
+        jsonSchema = factory.getJsonSchema(schemaObject);
+    }
+
+    public static boolean isMac() {
+        String OS = System.getProperty("os.name").toLowerCase();
+        return (OS.contains("mac"));
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        RepairnatorConfig.deleteInstance();
+        GitHelper.deleteFile(tmpDir);
+    }
+
+    @Test
+    public void testBearsJsonFileWithPassingPassingBuilds() throws IOException, ProcessingException {
+        long buggyBuildCandidateId = 386337343; // https://travis-ci.org/fermadeiral/test-repairnator-bears/builds/386337343
+        long patchedBuildCandidateId = 386348522; // https://travis-ci.org/fermadeiral/test-repairnator-bears/builds/386348522
+
+        tmpDir = Files.createTempDirectory("test_bears_json_file_passing_passing_builds").toFile();
+
+        Build buggyBuildCandidate = this.checkBuildAndReturn(buggyBuildCandidateId, false);
+        Build patchedBuildCandidate = this.checkBuildAndReturn(patchedBuildCandidateId, false);
+
+        BuildToBeInspected buildToBeInspected = new BuildToBeInspected(buggyBuildCandidate, patchedBuildCandidate, ScannedBuildStatus.PASSING_AND_PASSING_WITH_TEST_CHANGES, "test");
+
+        RepairnatorConfig config = RepairnatorConfig.getInstance();
+        config.setLauncherMode(LauncherMode.BEARS);
+
+        ProjectInspector4Bears inspector = new ProjectInspector4Bears(buildToBeInspected, tmpDir.getAbsolutePath(), null, null);
+        inspector.run();
+
+        // check bears.json against schema
+
+        JsonNode bearsJsonFile = jsonMapper.readTree(new File(inspector.getRepoToPushLocalPath() + "/bears.json"));
+
+        ProcessingReport report = jsonSchema.validate(bearsJsonFile);
+
+        String message = "";
+        for (ProcessingMessage processingMessage : report) {
+            message += processingMessage.toString()+"\n";
+        }
+        assertTrue(message, report.isSuccess());
+
+        // check correctness of the properties
```

Comment:
I don't follow you.

Reply:
Instead of doing the whole bunch of assert below, you could just put the expected JSON file in the resources of the test, get a JSON object from it using a parser, and compare the expected  JSON object against the created one.

Reply:
That will be great. I will try to do this, but I'm not that sure it will be just simple as you said, since there are some properties that the value can't be predicted.

Reply:
Like which value?

Reply:
Like dates and durations of the reproduction.

Reply:
I don't follow you there: you specify the build to get, then its information should never change anymore.

Reply:
Oh. Sorry. I just got it.

Reply:
I'm trying with this: https://github.com/skyscreamer/JSONassert
Let's see if I get this or not.

Reply:
Indeed. Maybe a middle-way solution could be to create the json file, to parse it and get the JSON object, and to change the dates and durations in this JSON object before making the assert

Reply:
I was thinking more in a way to "ignore" those values that can be different. I still don't know how to do. I'm just testing here now some stuff to see what this library returns, and then to see which approach I will take.

Reply:
No I think this is all

Reply:
Ok. I'm still on it.

Reply:
Done. I will wait for the build on my fork.